ddtrace/settings.py

```diff
@@ -103,6 +104,55 @@ def __repr__(self):
         return '{}.{}({})'.format(cls.__module__, cls.__name__, integrations)
 
 
+class IntegrationConfigItem(object):
+    """
+    A wrapper for items stored in an IntegrationConfig which is responsible for
+    determining what the value of the item stored in the config should be.
+
+    The precedence for determining the value for a config item is:
+
+    [high]
+        1) user-overridden value (indicated via `update_value`)
+        2) environment variable  (of the form DD_{INTEGRATION}_{KEY})
+        3) default value         (specified in the initializer)
+    [low]
+
+    TODO:
```

Comment:
I left these for subsequent PRs. They are fairly trivial to implement but would add some bloat to this PR. Thoughts on these `TODO`s?